jsapi/src/layermanager.js

```diff
@@ -0,0 +1,68 @@
+goog.provide('lux.LayerManager');
+
+goog.require('goog.dom');
+goog.require('ol.control.Control');
+
+/**
+ * @constructor
+ * @extends {ol.control.Control}
+ * @param {olx.control.ControlOptions} options Layer manager options.
+ * @export
+ */
+lux.LayerManager = function(options) {
+  var element = goog.dom.createDom('UL');
+  element.classList.add('lux-layer-manager');
+
+  ol.control.Control.call(this, {
+    element: element,
+    target: options.target
+  });
+};
+
+goog.inherits(lux.LayerManager, ol.control.Control);
+
+
+/**
+ * Set the map instance the control is associated with.
+ * @param {ol.Map} map The map instance.
+ */
+lux.LayerManager.prototype.setMap = function(map) {
+  ol.control.Control.prototype.setMap.call(this, map);
+  if (map) {
+    var layers = map.getLayers();
+    this.listenerKeys.push(
+        ol.events.listen(layers, ol.CollectionEventType.ADD,
+            this.update, this),
+        ol.events.listen(layers, ol.CollectionEventType.REMOVE,
+            this.update, this)
+    );
+  }
+};
+
+
+/**
+ * Update the component adequately.
+ */
+lux.LayerManager.prototype.update = function() {
+  goog.dom.removeChildren(this.element);
+  var layers = this.getMap().getLayers().getArray().slice().reverse();
+  layers.forEach(function(layer) {
+    var li = document.createElement('li');
+    var input = document.createElement('input');
+    var id = 'lux_' + goog.getUid(this) + '_' + goog.getUid(layer);
+    input.type = 'checkbox';
+    input.id = id;
+    input.checked = layer.get('visible');
+    input.onchange = function(e) {
+      layer.set('visible', e.target.checked);
+    };
+    li.appendChild(input);
```

Comment:
Maybe better wrap input into label to avoid id generation & simplify this a bit?